### ISSUE TITLE
Using d3.format("s") produces ugly results on axes that cross 0

### ISSUE THREAD

**Issue by teamdandelion** [d3/d3#1722]
Consider an axis with the following ticks:
    [-0.6000000000000001, -0.4000000000000001, -0.2000000000000001,   -8.881784197001253e-17, 0.1999999999999999, 0.3999999999999999, 0.5999999999999999, 0.7999999999999999, 1, 1.2, 1.4, 1.6, 1.8, 2]

I want to format them using SI-prefix formatting (to handle possible large numbers), so I use the formatter d3.format(".3s").

Unfortunately, the tiny-zero representation results in the following tick labels being generated: 
    ["-0.600", "-0.400", "-0.200", "-0.0888f", "0.200", "0.400", "0.600", "0.800", "1.00", "1.20", "1.40", "1.60", "1.80", "2.00"]

The -0.0888f representation looks very out of place. I understand why the algorithm generates this result, and looking at -8.881784197001253e-17 in isolation it makes sense to produce "-0.0888f" as the result. But in context, it's wrong. I believe this issue will only occur in cases very close to zero (otherwise the floating point error is dominated by a non-error component). So maybe the right way to fix this is to have the axis tick generator clean near-zero values as a special case.

**Comment by avipersin** [comment]
I am having a related problem using d3.v3*.js. When using high precision floating point numbers I am getting "-0.0" as a tick label instead of "0".
```
var x = d3.scale.linear()
    .domain([-.7, 2]);

// ["-0.6",
//  "-0.4",
//  "-0.2",
//  "-0.0",
//  "0.2",
//  "0.4",
//  "0.6",
//  "0.8",
//  "1.0",
//  "1.2",
//  "1.4",
//  "1.6",
//  "1.8",
//  "2.0"]
x.ticks(10).map(x.tickFormat(10));
```
Perhaps an edge case should be introduced to prevent "-0.0" as a tick label. Although mathematically correct it is a bit strange visually.  

```d3_locale_numberFormat(locals){...} ``` seems to be the culprit as it deals with the negative sign before applying the rounding. This may have been fixed in D3 v4 as I cannot find that function anymore but we are still using D3 v3.

**Comment by mbostock** [comment]
@avipersin Yes, there was a bug here that has been fixed in D3 4+; see d3/d3-format#14. There will not be any future D3 3.x releases, so you’ll either need to work around the bug in 3.x or upgrade to 4.x.
